### ISSUE TITLE
test_arange_dynamic doesn't work with capture scalar outputs

### ISSUE THREAD

**Issue by ezyang** [pytorch/pytorch#114139]
### 🐛 Describe the bug

This pre-existing test:

```
    def test_arange_dynamic(self, device):
        def fn(a):  
            batch_size = a.numel()
            max_len = a.max()
            return ~(
                torch.arange(0, max_len, device=a.device)
                .type_as(a)
                .repeat(batch_size, 1)
                .lt(a.unsqueeze(1))
            )
        
        a = torch.randint(10, 30, (10,), device=device)
        a[0] = 29  # fix max_len
        opt = self.compile_fn(fn)
        res = opt(a)
        ref = fn(a)
        self.assertEqual(res, ref)
```

fails when you set `capture_dynamic_output_shape_ops=True`.

A partial fix is https://github.com/pytorch/pytorch/pull/114137

But then you get

```
  File "/data/users/ezyang/a/pytorch/torch/_inductor/compile_fx.py", line 394, in compile_fx_inner
    compiled_graph = fx_codegen_and_compile(
  File "/data/users/ezyang/a/pytorch/torch/_inductor/compile_fx.py", line 605, in fx_codegen_and_compile
    graph.run(*example_inputs)
  File "/data/users/ezyang/a/pytorch/torch/_dynamo/utils.py", line 244, in time_wrapper
    r = func(*args, **kwargs)
  File "/data/users/ezyang/a/pytorch/torch/_inductor/graph.py", line 444, in run
    return super().run(*args)
  File "/data/users/ezyang/a/pytorch/torch/fx/interpreter.py", line 138, in run
    self.env[node] = self.run_node(node)
  File "/data/users/ezyang/a/pytorch/torch/_inductor/graph.py", line 782, in run_node
    result = ir.ExternKernel.require_stride_order(result, stride_order)
  File "/data/users/ezyang/a/pytorch/torch/_inductor/ir.py", line 3567, in require_stride_order
    assert is_stride_order_storage_and_layout(x, order)
  File "/data/users/ezyang/a/pytorch/torch/_inductor/ir.py", line 1589, in is_stride_order_storage_and_layout
    return layout.is_stride_ordered(stride_order)
  File "/data/users/ezyang/a/pytorch/torch/_inductor/ir.py", line 2270, in is_stride_ordered
    stride_ordered[order[i]] = V.graph.sizevars.size_hint(stride[i])
  File "/data/users/ezyang/a/pytorch/torch/_inductor/sizevars.py", line 389, in size_hint
    return int(out)
  File "/home/ezyang/local/a/pytorch-env/lib/python3.10/site-packages/sympy-1.12-py3.10.egg/sympy/core/expr.py", line 320, in __int__
    raise TypeError("Cannot convert symbols to int")
torch._dynamo.exc.BackendCompilerFailed: backend='inductor' raised:
TypeError: Cannot convert symbols to int
```

This error is similar to https://github.com/pytorch/pytorch/pull/113170

### Versions

main

cc @msaroufim @wconstab @bdhirsh @anijain2305 @zou3519 @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

**Comment by eellison** [comment]
Seems like it will be resolved by #113170

**Comment by ezyang** [comment]
works now
